src/imagery/i.eodag/testsuite/test_eodag.py

```diff
@@ -0,0 +1,435 @@
+            output="durham",
+            overwrite=True,
+        )
+        cls.peps_up = True
+        cls.cop_dataspace_up = True
+        cls.creodias_up = True
+        cls.planetary_computer_up = True
+        try:
+            cls.runModule(
+                "i.eodag",
+                flags="l",
+                provider="peps",
+                producttype="S2_MSI_L1C",
+                start="2024-01-01",
+                end="2024-01-01",
+                map="durham",
+                quiet=True,
+            )
+        except:
+            print(
+                "Faild to connect to provider 'peps', all testes associated with 'peps' will pass unconditionally."
+            )
+            cls.peps_up = False
+        try:
+            cls.runModule(
+                "i.eodag",
+                flags="l",
+                provider="cop_dataspace",
+                producttype="S2_MSI_L1C",
+                start="2024-01-01",
+                end="2024-01-01",
+                map="durham",
+                quiet=True,
+            )
+        except:
+            print(
+                "Faild to connect to provider 'cop_dataspace', all testes associated with 'cop_dataspace' will pass unconditionally."
+            )
+            cls.cop_dataspace_up = False
+        try:
+            cls.runModule(
+                "i.eodag",
+                flags="l",
+                provider="creodias",
+                producttype="S2_MSI_L1C",
+                start="2024-01-01",
+                end="2024-01-01",
+                map="durham",
+                quiet=True,
+            )
+        except:
+            print(
+                "Faild to connect to provider 'creodias', all testes associated with 'creodias' will pass unconditionally."
+            )
+            cls.creodias_up = False
+        try:
+            cls.runModule(
+                "i.eodag",
+                flags="l",
+                provider="planetary_computer",
+                producttype="LANDSAT_C2L2",
+                start="2024-01-01",
+                end="2024-01-01",
+                map="durham",
+                quiet=True,
+            )
+        except:
+            print(
+                "Faild to connect to provider 'planetary_computer', all testes associated with 'planetary_computer' will pass unconditionally."
+            )
+            cls.planetary_computer_up = False
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete temporary region settings."""
+        cls.del_temp_region()
+
+    def test_search_without_date(self):
+        """Test search without specifying dates."""
+        if not self.__class__.creodias_up:
+            return
+        self.assertModule(
+            "i.eodag",
+            flags="l",
+            provider="creodias",
+            producttype="S2_MSI_L2A",
+            clouds=30,
+            map="durham",
+            quiet=True,
+        )
+
+    def test_search_S2_MSI_L2A(self):
```

Comment:
This is how you could use _skipIf_:
```suggestion
    @unittest.skipIf(not available_providers["creodias"], "Provider 'creodias' is unavailable.")
    def test_search_S2_MSI_L2A(self):
```